projects/plugins/jetpack/modules/contact-form/grunion-contact-form.php

```diff
@@ -4147,33 +4195,46 @@ public static function get_mail_content_type() {
 	 *
 	 * This helps to ensure correct parsing by clients, and also helps avoid triggering spam filtering rules
 	 *
+	 * @param string $title - title of the email.
+	 * @param string $response_link - the link to the response.
+	 * @param string $form_link - the link to the form.
 	 * @param string $body - the message body.
+	 * @param string $footer - the footer containing meta information.
 	 *
 	 * @return string
 	 */
-	public static function wrap_message_in_html_tags( $body ) {
+	public static function wrap_message_in_html_tags( $title, $response_link, $form_link, $body, $footer ) {
 		// Don't do anything if the message was already wrapped in HTML tags
 		// That could have be done by a plugin via filters
 		if ( false !== strpos( $body, '<html' ) ) {
 			return $body;
 		}
 
+		$template = '';
+
+		/**
+		 * Filter the filename of the template HTML surrounding the response email. The PHP file will return the template in a variable called $template.
+		 *
+		 * @module contact-form
+		 *
+		 * @since $$next-version$$
+		 *
+		 * @param string the filename of the HTML template used for response emails to the form owner.
+		 */
+		require apply_filters( 'jetpack_forms_respone_email_template', __DIR__ . '/grunion-response-email-template.php' );
 		$html_message = sprintf(
 			// The tabs are just here so that the raw code is correctly formatted for developers
 			// They're removed so that they don't affect the final message sent to users
 			str_replace(
 				"\t",
 				'',
-				'<!doctype html>
-				<html xmlns="http://www.w3.org/1999/xhtml">
-				<body>
-
-				%s
-
-				</body>
-				</html>'
+				$template
 			),
-			$body
+			$title,
+			$body,
+			$response_link,
+			$form_link,
```

Comment:
Should we do some escaping here?

Reply:
I don't think we can escape $body, $footer or $template easily as they are supposed to have HTML. The two links can be escaped with esc_url() however.

Reply:
wp_kses with an extensive list of HTML tags?

Reply:
BTW - the response (in $body) has already been escaped and stripped of HTML tags before it reaches here.

Reply:
I'll esc_html() the title, esc_url() the two links, but the footer is made up of escaped strings. The body is the same.